src/test/scala/chisel3/tests/AsyncResetNativeTest.scala

```diff
@@ -0,0 +1,171 @@
+// See LICENSE for license details.
+
+package chisel3.tests
+
+import chisel3._
+import chisel3.experimental.MultiIOModule
+import chisel3.tester._
+import chisel3.tester.experimental.TestOptionBuilder._
+import chisel3.tester.internal.VerilatorBackendAnnotation
+import firrtl.AnnotationSeq
+import org.scalatest._
+
+/**
+  * circuit that illustrates usage of async register
+  * @param resetValue value on reset
+  */
+class NativeAsyncResetRegModule(resetValue: Int) extends MultiIOModule {
+  //noinspection TypeAnnotation
+
+  val io = IO(new Bundle {
+    val in = Input(UInt(8.W))
+    val out = Output(UInt(8.W))
+  })
+  val asyncReset = reset.asAsyncReset
+
+  val reg = withClockAndReset(clock, asyncReset) {
+    RegInit(resetValue.U(8.W))
```

Comment:
Consider doing the combinational assign to io.out in this block, then you don't need to pass a return to withClockAndReset. The register update probably also makes more sense inside the withClockAndReset scope.

Also, can you replace withClockAndReset with just withReset?

Reply:
done for all similar cases